src/Microsoft.PowerShell.Commands.Utility/commands/utility/CsvCommands.cs

```diff
@@ -883,16 +888,36 @@ internal static IList<string> BuildPropertyNames(PSObject source, IList<string>
                 throw new InvalidOperationException(CsvCommandStrings.BuildPropertyNamesMethodShouldBeCalledOnlyOncePerCmdletInstance);
             }
 
-            // serialize only Extended and Adapted properties..
-            PSMemberInfoCollection<PSPropertyInfo> srcPropertiesToSearch =
-                new PSMemberInfoIntegratingCollection<PSPropertyInfo>(
+            propertyNames = new Collection<string>();
+            if (source.BaseObject is IDictionary dictionary)
```

Comment:
It was my question in original issue - why do we consider only IDictionary? If we want to change a design of the cmdlet we should support _all_ possible variants.